scripts/sortTermsAlphabetically.gradle

```diff
@@ -0,0 +1,108 @@
+import static groovy.io.FileType.FILES
+import java.util.regex.Matcher
+
+class Term {
+    String term
+    String captionEN
+    String captionDE
+    String includeFragment
+
+    Term(String term, String captionEN, String captionDE, includeFragment) {
+        this.term = term
+        this.captionEN = captionEN
+        this.captionDE = captionDE
+        this.includeFragment = includeFragment
+    }
+}
+
+task sortTermsAlphabetically(
+        description: 'collect terms and create an alphabetically orders structure',
+        group: 'isaqb-glossary'
+) {
+    doLast {
+        def terms = createTermsContent()
+
+        sortTermsEN(terms)
+        writeTermsStructure(terms, "EN")
+
+        sortTermsDE(terms)
+        writeTermsStructure(terms, "DE")
+    }
+
+    ext.createTermsContent = {
+        def terms = collectTerms()
+        List alphabet = ('A'..'Z').collect { it }
+        alphabet.each(
+                letter -> terms.add(new Term(/==== $letter/, /==== $letter/, /==== $letter/, /=== $letter/))
+        )
+        return terms
+    }
+
+    ext.collectTerms = {
+        def termsFolder = new File(projectDir, '/docs/1-terms')
+        def terms = []
+
+        termsFolder.traverse(type: FILES) { file ->
+
+            if (file.name ==~ 'term.*[.](ad|adoc|asciidoc)$') {
+                println "Learning term from file: " + file.getAbsolutePath()
+                def content = file.text
+
+                def term
+                def refPath = termsFolder.toPath().relativize( file.toPath() ).toFile()
+                def includeFragment = "include::${refPath}[{include_configuration}]";
+                def caption_EN
+                def caption_DE
+
+                if (content =~ ~/.*(\[#.*\]).*/) {
+                    term = Matcher.lastMatcher[0][1]
+
+                }
+                def languageCode;
+                languageCode = 'EN'
+                if (content =~ ~('(?msi)TAG::'+languageCode+'\\[\\].*?(^====.*?$).*?END::'+languageCode+'\\[\\]') ) {
```

Comment:
Now that's amazing magic 😂  Maybe we can add some documentation that explains what happens? Maybe at the top of the file as class comment?

Reply:
Sure.